server/engine/src/main/java/org/eclipse/lsp/cobol/implicitDialects/cics/utility/CICSReceiveOptionsCheckUtility.java

```diff
@@ -63,90 +64,47 @@ private void checkGroupOne(CICSParser.Cics_receive_group_oneContext ctx) {
     if (!ctx.ASIS().isEmpty() || !ctx.BUFFER().isEmpty()) {
       checkHasIllegalOptions(ctx.LEAVEKB(), "LEAVEKB");
     }
-
-    List<RuleContextData> contexts = new ArrayList<>();
-    contexts.add(new RuleContextData(ctx.cics_into_set(), "INTO or SET"));
-    contexts.add(new RuleContextData(ctx.cics_maxlength(), "MAXLENGTH or MAXFLENGTH"));
-    contexts.add(new RuleContextData(ctx.ASIS(), "ASIS", ErrorSeverity.WARNING));
-    contexts.add(new RuleContextData(ctx.BUFFER(), "BUFFER", ErrorSeverity.WARNING));
-    contexts.add(new RuleContextData(ctx.NOTRUNCATE(), "NOTRUNCATE", ErrorSeverity.WARNING));
-    contexts.add(new RuleContextData(ctx.LEAVEKB(), "LEAVEKB", ErrorSeverity.WARNING));
-    contexts.add(new RuleContextData(ctx.cics_handle_response(), "RESPONSE HANDLER"));
-    harvestResponseHandlers(ctx.cics_handle_response(), contexts);
-    checkDuplicates(contexts);
+    checkResponseHandlers(ctx.cics_handle_response());
+    checkDuplicates(ctx, "ASIS", "BUFFER", "NOTRUNCATE", "LEAVEKB");
```

Comment:
Is there a case when one of these keywords triggers warning in one rule, but error in another?
If not, we could have a single static map.
If so, we could still have a handful of static maps.

Reply:
The benefit would be exactly not creating the hashmap everytime the function is called.